ext/vulnsrc/redhat/redhat.go

```diff
@@ -0,0 +1,731 @@
+	// use criteria parse result
+	moduleNamespaces := ParseCriteriaForModuleNamespaces(advisoryDefinition.Criteria)
+	if len(moduleNamespaces) > 0 {
+		// use MODULE namespace
+		return moduleNamespaces[0]
+	} else {
+		// use CPE namespace
+		cpeNames, err := ParseCpeNamesFromAffectedCpeList(advisoryDefinition.Metadata.Advisory.AffectedCpeList)
+		if err != nil {
+			// log error and continue
+			log.Error(err)
+			return ""
+		}
+		if len(cpeNames) == 0 {
+			// no namespace found
+			return ""
+		} else {
+			return cpeNames[0]
+		}
+	}
+}
+
+func GetSeverity(severity string) database.Severity {
+	switch strings.Title(strings.ToLower(severity)) {
+	case "None":
+		return database.NegligibleSeverity
+	case "Low":
+		return database.LowSeverity
+	case "Moderate":
+		return database.MediumSeverity
+	case "Important":
+		return database.HighSeverity
+	case "Critical":
+		return database.CriticalSeverity
+	default:
+		log.Warningf("could not determine vulnerability severity from: %s.", severity)
+		return database.UnknownSeverity
+	}
+}
+
+// checks whether the given severity is significant (used to determine whether vulns will be parsed and stored for it)
+func IsSignificantSeverity(severity string) bool {
+	switch strings.Title(strings.ToLower(severity)) {
+		case "None":
+			return false
+	default:
+		// anything else is considered significant
+		return true
+	}
+}
+
+func extractAllCriterions(criteria OvalV2Criteria) []OvalV2Criterion {
+    var criterions []OvalV2Criterion
+    for _, criterion := range criteria.Criteria {
+        // recursively append criteria contents
+        criterions = append(criterions, extractAllCriterions(criterion)...)
+    }
+    for _, criterion := range criteria.Criterion {
+        // append criterion
+        criterions = append(criterions, criterion)
+    }
+    return criterions
+}
+
+func ParseNVRA(rpmName string) RpmNvra {
+	// parse NVRA from RPM name
+	// golang-1.6.3-2.el7.x86_64.rpm
+	// Name        : golang
+	// Version     : 1.6.3
+	// Release     : 2.el7
+	// Architecture: x86_64
+	var regexRpmNVRA = regexp.MustCompile(`(.*/)*(.*)-(.*)-(.*?)\.([^.]*)(\.rpm)`)
+	matches := regexRpmNVRA.FindStringSubmatch(rpmName)[2:6]
+	rpmNvra := RpmNvra{matches[0], matches[1], matches[2], matches[3]}
+	return rpmNvra
+}
+
+func IsRmpArchSupported(rpmName string) bool {
+	return IsArchSupported(ParseNVRA(rpmName).Arch)
+}
+
+func IsArchSupported(arch string) bool {
+	return SupportedArches[arch]
+}
+
+// parse affected_cpe_list (first entry from CPE list should not be used because it doesn't come from Advisory configuration)
+func ParseCpeNamesFromAffectedCpeList(affectedCpeList OvalV2Cpe) ([]string, error) {
+	var cpeNames []string
+	if affectedCpeList.Cpe == nil || len(affectedCpeList.Cpe) < 2 {
+		return cpeNames, errors.New("unparseable affected cpe list")
+	}
+	// parse and return any entries after the first cpe entry from the list
+	for i := 1; i < len(affectedCpeList.Cpe); i++ {
```

Comment:
I looked at ovalv2 and we can't tell which CPE is useful and which is not so please don't skip cpe[0] and use them all.